envsec/internal/auth/aws.go

```diff
@@ -0,0 +1,155 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/MicahParks/keyfunc/v2"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cognitoidentity"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/golang-jwt/jwt/v5"
+	"go.jetpack.io/envsec/debug"
+)
+
+// for valiation of JWT (can be hardcoded to avoid fetching)
+const jwksURL = `https://jetpack-io.us.auth0.com/.well-known/jwks.json`
+const awsRegion = "us-east-1"
+
+// have to be var instead of const so that their &address can be passed
+var accountId = "984256416385"
+var identityPoolId = "us-east-1:da3c3c71-61c7-4f7c-8e3d-3770e9b61379"
+
+// Data that will be in JWT
+type UserClaim struct {
+	jwt.RegisteredClaims
+	OrgID string `json:"https://auth.jetpack.io/org_id,omitempty"`
+}
+
+// getting the saved auth token after login
+func getAuthToken() (string, error) {
+	type auth struct {
+		AccessToken  string `json:"access_token"`
+		RefreshToken string `json:"refresh_token"`
+		IdToken      string `json:"id_token"`
+		TokenType    string `json:"token_type"`
+		ExpiresIn    int    `json:"expires_in"`
+	}
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	authFile, err := os.Open(fmt.Sprintf("%s/.local/state/devbox/auth.json", homedir))
+	if err != nil {
+		return "", err
+	}
+	defer authFile.Close()
+
+	byteContent, err := io.ReadAll(authFile)
+	if err != nil {
+		return "", err
+	}
+
+	var result auth
+	json.Unmarshal([]byte(byteContent), &result)
+	return result.AccessToken, nil
+}
```

Comment:
The auth library has a function that already provides the user and tokens, handles login/refresh, etc.